src/background/Wallet/Wallet.ts

```diff
@@ -899,10 +939,30 @@ export class Wallet {
     this.emitter.emit('chainChanged', chain, origin);
   }
 
+  /** A helper for interpretation in UI */
+  async uiGetEip712Transaction({
+    params: { transaction },
+    context,
+  }: WalletMethodParams<{ transaction: IncomingTransactionWithChainId }>) {
+    this.verifyInternalOrigin(context);
+
+    const { chainId } = transaction;
+    const prepared = prepareTransaction(transaction);
+    const typedData = createTypedData({
+      chainId: normalizeChainId(chainId),
+      message: EIP712Signer.getSignInput(prepared),
+    });
+    return typedData;
+  }
+
   private async getProvider(chainId: ChainId) {
     const networks = await networksStore.loadNetworksWithChainId(chainId);
     const nodeUrl = networks.getRpcUrlInternal(networks.getChainById(chainId));
-    return new ethers.providers.JsonRpcProvider(nodeUrl);
+    if (FEATURE_PAYMASTER_ENABLED) {
+      return new ZksProvider(nodeUrl);
```

Comment:
Will we use this provider for all chains in the future?

Reply:
I haven't decided whether to keep using the zksync lib at all, but as I understand the provider can be safely used for regular transactions, too. The current branching is more cautionary than practical